### DISCUSSION TITLE
Zen Web panel customization

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#852] [Ideas]
### What feature would you like to see?

I love the look and animations of the web panel, as well as toggling between floating and side by side viewing. I would love to be able to have that be its own sidebar, the way floorp does it. I often put my most used applications/communications there, so having instant access really important. Also, having notifications reflected in the favicons - for example a badge with the number or unread emails, or new messages, would be amazing. 

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/370_

**Comment by marcosjrvrael** [comment]
I want to be able to move the panel in configuration menu to the other browser side.

**Comment by blaszta** [comment]
1. Enable add-ons (especially Dark Reader) in web panel. Wikipedia page is too bright!
2. Ability to edit favicon/create title (maybe generic icon with abbreviation of the title?) of a web panel. I have Uptime Kuma to monitor all my development server/database/etc. I think this is a perfect use for web panel, I can monitor everything at a glance.

**Comment by AntonioKowatsch** [comment]
Same here. Just like @marcosjrvrael mentioned. 
I would like to move the web panel to the right hand side of the browser while keeping the vertical tabs on the left side.
I don't like the look of having the vertical tabs and the web panel occupy the same side.

**Comment by Frai7ty** [comment]
I also would love to be able to have Web Panels on the right hand side.

**Reply by mr-cheffy** [reply]
https://zen-browser.app/themes/dd4f5461-1564-4e56-9f9d-f81e3c18f93c

**Reply by SalahAdDin** [reply]
> https://zen-browser.app/themes/dd4f5461-1564-4e56-9f9d-f81e3c18f93c

It does not work in the latest Zen Browser version.

**Reply by Bliwi** [reply]
@SalahAdDin It does work. It just comes disabled by default, you need to open the zen-addon settings and enable it.

**Comment by GuMarques** [comment]
The ability to use the panels in float mode would rock, something similar to what Opera does.

**Comment by ThatAmuzak** [comment]
Also the ability to modify the max width of the web panel would be great!

**Reply by Axenide** [reply]
Supposedly there is `zen.view.sidebar-expanded.max-width` in `about:config` but it doesn't work. I submitted a theme for this and Mauro told me that this exists. I hope he fixes it soon. :)
Meanwhile, this is the CSS I made for it:
```css
#zen-sidebar-web-panel {
  max-width: 1000px !important;
}
```
Add it to your `userChrome.css`. :D

**Reply by ThatAmuzak** [reply]
oh wow, works like a charm! Thank you very much!

**Reply by DariusSlowik** [reply]
where do I find userChrome.css?

**Reply by Axenide** [reply]
> where do I find userChrome.css?

In your profile directory, inside the chrome folder. If you don't have it you can create it.

**Comment by josephhartBRD** [comment]
The web panel used to be split, and i liked that. Now its floating, would be good to be able to toggle floating on/off.

**Reply by GuMarques** [reply]
Agree, also in the new float mode I wasn't able to resize it and some websites doesn't work properly in a small window.

**Reply by Axenide** [reply]
What do you mean? It can be toggled.

https://github.com/user-attachments/assets/564adf35-688e-4029-b958-8106312d8113

**Comment by josephhartBRD** [comment]
Oh thank you!



Joseph Hart (pronounced: JOH-sef HART) | He/Him<https://pronouns.org/he-him>
Director, InFocus Platform & Experience Studio, DCoE | Broadridge Financial Solutions, Inc.


 [cid:3cd66855-7eda-4d36-84a3-20d8912291d4] <https://www.linkedin.com/company/broadridge-financial-solutions>  [cid:c9c6b023-f79d-494b-8737-dcbb221c338f] <https://twitter.com/broadridge>  [cid:1ff7cc42-edb8-44a0-a729-c773eb12d149] <https://www.youtube.com/user/BroadridgeInsights>  [cid:28be1d4f-2e51-4021-a83d-5751a6972254] <https://www.facebook.com/BroadridgeCareers>

broadridge.com<http://www.broadridge.com/>







________________________________

**Comment by tedbergstrand** [comment]
I love the web panel. It's my favorite part of Zen Browser. And I love that you want to make it better.

1.) Zoom control
  - The text is pretty small on a 4k monitor at a distance. Being able to blow it up would be awesome.

2.) Cursor Detection for Forward/Back
  - It's ingrained in most of us to use the back and forward buttons on our mouse. But when you're working in a web panel, it's easy to forget that you're not in a real tab. And it's slightly frustrating to send the main tab backwards instead of the web panel. So maybe if the cursor is within the web-panel, apply the action to the panel. If it's in the tab frame, apply it to the tab.
  
3.) Increased size for web-panel shortcut buttons.
  - You can Ctrl+Tab through tabs, or expand the width for a bigger target area for the normal buttons, but the webpanel buttons have to be clicked. And they're pretty small on 4k screens. I think they should be at least the size of the cursor.

**Comment by 12th-devs** [comment]
Have the option for it to run in the background, So I can use it as a music player sometimes.

**Reply by Axenide** [reply]
It runs in the background, but recently there's a bug that reloads the panels. Though it has been found and should be fixed soon.

**Comment by lvickyj** [comment]
imo the top and bottom parts of the web panel are a little too large, so it would be great to have options to hide or to make it smaller. It would be nice to have adjustable zoom in the panel and have the current page return after it auto hides when you click somewhere else

**Comment by me-nkr** [comment]
I would say keyboard navigation in web pane wouldl be something useful

**Comment by devSaifur** [comment]


**Reply by cheeeebs** [reply]
You can do this already by clicking the minus (-) button on the toolbar of the web panel. 

![image](https://github.com/user-attachments/assets/1266eb65-e9dc-44d8-857f-6963bd234b02)

**Reply by devSaifur** [reply]
Yes, but its splits the main page.  If I want to split two-page side by side I can just use tab splitting. Web panel should stay over the main page not split it. Just like it stays when scrolling but gets minimized on click.

**Reply by cheeeebs** [reply]
You can go to `Settings>Look and Feel>Side Web Panels` and disable `Close the panel when it loses focus if it's not pinned`. This lets you float the web panels without closing when interacting with webpages.

**Reply by devSaifur** [reply]
Yeah that works! You just solved my problem. tnq u very much.

**Comment by DivineMK** [comment]
I would like to suggest including keyboard shortcuts for web panels navigation, either individually open the panels like Floorp as in #388 or for switching between web panels in the sidebar as suggested in this [comment](https://github.com/zen-browser/desktop/discussions/852#discussioncomment-11117534). This is possibly a duplicate of this [comment](https://github.com/zen-browser/desktop/discussions/852#discussioncomment-11195831), but with a better summary. This is also a feature I really miss coming from Floorp. Otherwise Zen performance is amazing

**Comment by drod3763** [comment]
I'd like to keep my vertical tabs on the left and pin the webpanel on the right side.

**Reply by Michael679089** [reply]
yep

**Comment by DetroNot74** [comment]
Would like a CONTEXT MENU option to OPEN/SEARCH into web-Panel.
Its cumbersome to do it manually.

**Comment by Michael679089** [comment]
Is there a way to move my zen web panels to the right instead of it being on the left?

I don't like the web panels being on the left, the same side where my tabs are, I would like it to be on the right.

**Reply by M1ke000** [reply]
Install this mod, then go to settings > Zen Mods > click the gear icon under the mod and enable it https://zen-browser.app/mods/dd4f5461-1564-4e56-9f9d-f81e3c18f93c/

**Reply by Sundvallen** [reply]
Doesn't seem to work when pinning the web panel

**Reply by drod3763** [reply]
Doesn't work for me either.

**Reply by encleine** [reply]
you can move the pinned web panel to the right with this snippet 
```css
#tabbrowser-tabbox > #zen-sidebar-web-panel-wrapper {
  order: 5 !important;
} 
``` 
I don't know if 5 works for all layouts but it works for the single toolbar

**Comment by p134c0d3** [comment]
I would like a setting to be able to choose what item in the web I want to open by default, as well as an option for it to open the last selected item. For example:
![image](https://github.com/user-attachments/assets/dc20557d-e059-46c8-a1c4-3e71e5324ed6)

If I want Telegram to be selected and opened by default when opening via the keyboard shortcut, or Discord, X, etc. Or if Telegram was the last open item when I closed the panel, when I open it again, the last open item (Telegram) would be selected and load by default.

**Reply by p134c0d3** [reply]
- Option A: Have a dropdown list of icons (current items user has pinned to web panel) user can select one to load by default when web panel is opened
- Option B: Have an option for the last open item to be loaded when web panel is opened.

**Comment by locrianz** [comment]
I love the floating web panel so much that I was disappointed to find I couldn't move it outside of the Zen Browser window. I have multiple monitors and would love to be able to move the panel to my other screen.

**Comment by Akiyamka** [comment]
How to enable web panel on current version of zen browser?

**Reply by Axenide** [reply]
It was removed due to it being unstable and hard to maintain.
